google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/BuiltinMetricsTracer.java

```diff
@@ -272,10 +292,16 @@ public void disableFlowControl() {
   }
 
   private void recordOperationCompletion(@Nullable Throwable status) {
+    if (operationFinishedEarly.get()) {
+      status = null; // force an ok
+    }
+
     if (!opFinished.compareAndSet(false, true)) {
       return;
     }
-    operationTimer.stop();
+    if (operationTimer.isRunning()) {
```

Comment:
I think it's unnecessary to check if the timer is still running and stop it, we can just do  `long operationLatencyNano = operationTimer.elapsed(TimeUnit.NANOSECONDS);`.  (sorry about my redundant code)